apps/settings/js/panels/call_forwarding/panel.js

```diff
@@ -0,0 +1,244 @@
+        };
+
+        for (var key in mapping) {
+          var rules = allRules[key];
+          var elementId = mapping[key];
+          var element = this._elements[elementId];
+          var rule = CallUtils.findActiveVoiceRule(rules);
+          if (rule) {
+            navigator.mozL10n.setAttributes(element,
+              'callForwardingForwardingVoiceToNumber', {
+                number: rule.number
+            });
+          } else {
+            element.setAttribute('data-l10n-id',
+              'callForwardingNotForwarding');
+          }
+        }
+      },
+
+      _bindMenuItemsClickEvent: function() {
+        // We will concate strings like (for grep use)
+        //
+        // ril.cf.mobilebusy.enabled
+        // ril.cf.mobilebusy.number
+        // ril.cf.noreply.number
+        // ril.cf.noreply.enabled
+        // ril.cf.notreachable.number
+        // ril.cf.notreachable.enabled
+        // ril.cf.unconditional.number
+        // ril.cf.unconditional.enabled
+        var mapping = {
+          mobileBusy: {
+            menuItem: this._elements.mobileBusyMenuItem,
+            panelId: 'call-cf-mobile-busy-settings',
+            settingsKey: 'mobilebusy'
+          },
+          noReply: {
+            menuItem: this._elements.noReplyMenuItem,
+            panelId: 'call-cf-no-reply-settings',
+            settingsKey: 'noreply'
+          },
+          notReachable: {
+            menuItem: this._elements.notReachableMenuItem,
+            panelId: 'call-cf-not-reachable-settings',
+            settingsKey: 'notreachable'
+          },
+          unconditional: {
+            menuItem: this._elements.unConditionalMenuItem,
+            panelId: 'call-cf-unconditional-settings',
+            settingsKey: 'unconditional'
+          }
+        };
+
+        var onclick = (panelId, settingsKey) => {
+          SettingsCache.getSettings((results) => {
+            var number = results['ril.cf.' + settingsKey + '.number'];
+            var enabled = results['ril.cf.' + settingsKey + '.enabled'];
+
+            // Reflect related settings value on the panel
+            DialogService.show(panelId, {
+              number: number,
+              enabled: enabled
+            }).then((result) => {
+              this._lockMenuItems(true);
+              var type = result.type;
+              var value = result.value || {};
+              var returnNumber = value.number || '';
+              var returnEnabled = value.enabled || false;
+
+              if (type === 'submit') {
+                var obj = {};
+                obj['ril.cf.' + settingsKey + '.number'] = returnNumber;
+                obj['ril.cf.' + settingsKey + '.enabled'] = returnEnabled;
+
+                var req = mozSettings.createLock().set(obj);
+                req.onsuccess = () => {
+                  this._updateAllDescInfoWithL10nId('callSettingsQuery');
+                  CallForwarding.setCallForwardingValues(this._cardIndex, {
```

Comment:
Based on our offline discussion, `CallForwarding`, `CallBarring` ... etc are modules that won't notice about the scheduling, what they do is just a function call communicating with Gecko -> modem and that's it. 

if we are going to handle this scheduling problems, developers have to make sure to call this function inside CallSettingsTaskScheduler.